userspace/libscap/engine/bpf/attached_prog.c

```diff
@@ -84,6 +84,7 @@ static int __attach_tp(struct bpf_attached_prog* prog, char* last_err)
 	attr.sample_period = 1;
 	attr.wakeup_events = 1;
 	attr.config = id;
+	attr.size = sizeof(struct perf_event_attr);
```

Comment:
I'm not sure we want to hardcode this value while we compile, maybe a new value is not supported by an old kernel :thinking: what is the default value if we leave it uninitialized?

Reply:
This does **exactly** what the man page states to do...

Reply:
https://linux.die.net/man/2/perf_event_open, quoting from the OP:
```
 size   The size of the perf_event_attr structure for
              forward/backward compatibility.  Set this using
              sizeof(struct perf_event_attr) to allow the kernel to see
              the struct size at the time of compilation.

              The related define PERF_ATTR_SIZE_VER0 is set to 64; this
              was the size of the first published struct.
              PERF_ATTR_SIZE_VER1 is 72, corresponding to the addition
              of breakpoints in Linux 2.6.33.  PERF_ATTR_SIZE_VER2 is 80
              corresponding to the addition of branch sampling in Linux
              3.4.  PERF_ATTR_SIZE_VER3 is 96 corresponding to the
              addition of sample_regs_user and sample_stack_user in
              Linux 3.7.  PERF_ATTR_SIZE_VER4 is 104 corresponding to
              the addition of sample_regs_intr in Linux 3.19.
              PERF_ATTR_SIZE_VER5 is 112 corresponding to the addition
              of aux_watermark in Linux 4.1.
```